test/DynamoCoreWpfTests/DynamoViewTests.cs

```diff
@@ -123,5 +124,78 @@ public void OpeningWorkspaceWithTrustWarning()
             Assert.IsFalse(ViewModel.FileTrustViewModel.ShowWarningPopup);
             DynamoModel.IsTestMode = true;
         }
+
+        [Test]
+        public void TestToastNotificationClosingBehavior()
+        {
+            var preferencesWindow = new PreferencesView(View);
+            preferencesWindow.Show();
+            DispatcherUtil.DoEvents();
+            string selectedLanguage = (string)((ComboBox)preferencesWindow.FindName("LanguageCmb")).SelectedItem;
+
+            bool? xisToastNotificationVisible = ViewModel.MainGuideManager?.ExitTourPopupIsVisible;
+            ViewModel.PreferencesViewModel.SelectedLanguage = selectedLanguage == "English" ? "Español" : "English";
+
+            DispatcherUtil.DoEvents();
+            Exit_FirstPart();
+
+            bool isToastNotificationVisible = (bool)(ViewModel.MainGuideManager?.ExitTourPopupIsVisible);
+            Assert.IsFalse(isToastNotificationVisible);
+
+            Exit_SecondPart();
+        }
+
+        /// <summary>
+        /// First part of the Exit in order to catch the viewModel info
+        /// </summary>
+        public void Exit_FirstPart()
+        {
+            //Ensure that we leave the workspace marked as
+            //not having changes.
+            ViewModel.HomeSpace.HasUnsavedChanges = false;
+
+            if (View.IsLoaded)
+                View.Close();
+
+            if (ViewModel != null)
+            {
+                var shutdownParams = new DynamoViewModel.ShutdownParams(
+                    shutdownHost: false, allowCancellation: false);
+
+                ViewModel.PerformShutdownSequence(shutdownParams);
+            }
+        }
+
+        /// <summary>
+        /// Second part of the Exit in order to catch the viewModel info
+        /// </summary>
+        public void Exit_SecondPart()
+        {
+            if (ViewModel != null)
+            {
+                ViewModel = null;
+            }
+
+            View = null;
+            Model = null;
+            preloader = null;
+
+            try
+            {
+                var directory = new DirectoryInfo(TempFolder);
+                directory.Delete(true);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.StackTrace);
+            }
+        }
+
+        /// <summary>
+        /// Overrides the Exit base class method to custom handling
+        /// </summary>
+        public override void Exit()
```

Comment:
I don't think this is a good idea.
You change the exit behavior of every test in this fixture and it will be unexpected for other developers.

instead of splitting the exit code up, perhaps you can just call `close` on the DynamoView and then see if your popup is still visible.

I would not spend too long on this, I prefer no test for this, to test code that will lead to hard to debug test failures later.

Reply:
I have updated the Test simplifying it, its feature passed locally (but throw error in the master-15 due to a different issue), even if it doesn't meet with your approval just let me know so I can close it. Thanks.

![image](https://github.com/DynamoDS/Dynamo/assets/96534278/cea508c8-09b5-4f73-9f60-2db850a7a6ef)


> I don't think this is a good idea. You change the exit behavior of every test in this fixture and it will be unexpected for other developers.
> 
> instead of splitting the exit code up, perhaps you can just call `close` on the DynamoView and then see if your popup is still visible, then make exit do some null checks.
> 
> I would not spend too long on this, **I prefer no test for this, to test code that will lead to hard to debug test failures later.**